lib/errorUtils.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+const Assert = require('assert')
+
+const types = {
+  system: [
+
+      // JavaScript
+
+    EvalError,
+    RangeError,
+    ReferenceError,
+    SyntaxError,
+    TypeError,
+    URIError,
+
+      // Node
+
+    Assert.AssertionError
+  ]
+}
+
+function reflectSystemErrors (err) {
+  for (let i = 0; i < types.system.length; ++i) {
+    if (err instanceof types.system[i]) {
```

Comment:
can you avoid the property lookup?

Reply:
There is only one per iteration. What do you mean?

Reply:
I think he means storing the array as its own variable so you don't have to access the `system` property on the `types` object on every iteration. The `types` object is sort of unnecessary since it only has one property, so you could just store the array as something like `const systemTypes = [...]` at the top level.

Reply:
Such micro-optimization is really necessary for an error case? :/